### PR TITLE
Crash when pressing B button on playlists #16951

### PR DESCRIPTION
**Pull request by jbreitweiser**
## Description
 This is a correction to a null value when the full image name is not populated correctly.  Added check to avoid null pointer error when checking for alternative file names.

## Related Issues

Crash when pressing B button on playlists
#16951

## Reviewers

[If possible @mention all the people that should review your pull request]

### CODE REVIEW

**Comment by zoltanvb** [comment]
Negation is missing (!string_is_empty), but unfortunately even after adding it, I can still reproduce the crash.
I suspect it is related to the loading procedure - the crash happens, when one moves away quickly from a playlist, I need to do left - backspace key sequence quickly one after the other, to make it happen reliably, though I have seen it happen in some other cases as well.

**Comment by LibretroAdmin** [comment]
Maybe it's for the best if that PR that introduced all this code gets reverted and you go back to the drawing board on this stuff and make sure these crashes no longer occur on the next PR? That might be for the best vs. having to keep building on a broken foundation.

**Comment by jbreitweiser** [comment]
@zoltanvb do you have icons in the logos thumbnail directory.  I want to
replicate your issue.  If so what are the file names.  I am going to back
track to the population of the path object to see if that img full name
variable in your stack trace might be an issue.  Can you also give me the
playlist you are using.  Maybe that will help replicate the issue.

**Comment by zoltanvb** [comment]
I do have a few for testing, but not for the last playlist where the crash usually happens, I believe they are not coming into play. Anyway I attach the playlist, with fake paths.
[Sinclair - ZX Spectrum.zip](https://github.com/user-attachments/files/16919082/Sinclair.-.ZX.Spectrum.zip)

Image full name and image short name can be empty on purpose, to save some queries, it has been like that since a while.
Short name: https://github.com/libretro/RetroArch/blob/c0715fc420d232b651e26a55861f2d93b63db67d/gfx/gfx_thumbnail_path.c#L53
Full name: https://github.com/libretro/RetroArch/blob/c0715fc420d232b651e26a55861f2d93b63db67d/gfx/gfx_thumbnail_path.c#L499

**Comment by LibretroAdmin** [comment]
Alright @jbreitweiser , we'll give you the opportunity for now to track down this issue and resolve it. If that doesn't work we'll have to maybe consider the nuclear option of reverting it all, but for now you have the initiative.

**Comment by jbreitweiser** [comment]
@zoltanvb I still cant replicate the issue but I think the issue is de initialization of the tumbnails.  I reverted the null checks.  The error seems to be a null pointer caused by moving off the playlist to a none playlistlist.  So I updated populate playlist (called when changing categories or lists) to check if there are active downloads and them de-initialize them.  Please see if you can still get the error.  Thanks.

**Comment by zoltanvb** [comment]
@jbreitweiser I think push was missing, as I still see the previous commit.

**Comment by jbreitweiser** [comment]
@zoltanvb pushed now.

**Comment by zoltanvb** [comment]
Thank you! That does the trick, I can no longer reproduce the crash.

File: menu/drivers/xmb.c [hunk 1]
```diff
@@ -2958,9 +2958,14 @@ static void xmb_populate_entries(void *data,
...
 
-   if(xmb->is_playlist && settings->uints.menu_icon_thumbnails)
+   if(xmb->is_playlist)
    {
-      xmb_populate_dynamic_icons(xmb);
+      if(settings->uints.menu_icon_thumbnails)
+         xmb_populate_dynamic_icons(xmb);
+   } 
+   else if(xmb->thumbnails.pending_icons != XMB_PENDING_THUMBNAIL_NONE )
+   {
```
Comment (LibretroAdmin) [inline menu/drivers/xmb.c:2967]: See CODING-GUIDELINES, if a code block only has one line, then just drop the brackets

**Review by LibretroAdmin — CHANGES_REQUESTED** [review]

**Comment by LibretroAdmin** [comment]
Other than that nitpick, it looks fine.

You'll have to rebase the PR though since some changes were made since your last push.

**Review by viachaslavic — CHANGES_REQUESTED** [review]

**Comment by jbreitweiser** [comment]
code formatted per comments.

**Pull request merged**